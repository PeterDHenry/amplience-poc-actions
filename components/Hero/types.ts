export type HeroProps = {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  cssClass?: string
  background: {
    _meta: {
      schema: string
    }
    id: string
    name: string
    endpoint: string
    defaultHost: string
  }
  //color: string
  //contrastColor: string
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline?: string
  }
}

export const defaultProps: HeroProps = {
  _meta: {
    name: 'Ride like a pro banner',
    schema: 'https://wiggle.com/hero-banner',
    deliveryId: '18a7eee8-6c7a-4a92-8a76-29eb936679a8',
  },
  background: {
    _meta: {
      schema: 'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
    },
    id: 'ef717f48-99e2-4717-89da-6f906056eae1',
    name: 'CCC-11-5-a',
    endpoint: 'wigglecrcnonprd',
    defaultHost: 'cdn.media.amplience.net',
  },
  textAndCTA: {
    headline: 'Ride like a pro',
    callToActionURL: 'https://www.wiggle.co.uk/pro-riding-for-summer',
    callToActionText: 'Shop now',
  },
}
