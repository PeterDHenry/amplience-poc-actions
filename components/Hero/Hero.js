import React from 'react';

import HeroImage from '../LazyLoadImage/LazyLoadImage'

const Hero = ( {content} ) => (
  <a className="relative block group" href={content.calltoactionurl}>
    <HeroImage id={content.background.name} />
    <div className="absolute bottom-40 left-20 uppercase text-white max-w-md">
      <h3 className="text-7xl font-bold">
        {content.headline}
      </h3>
      <h2 className="text-3xl font-bold">
        {content.strapline}
      </h2>
      <div className="max-w-xs border-solid border-2 border-white p-1 group-hover:bg-white group-hover:text-black">
        {content.calltoactiontext}
      </div>
    </div>
  </a>
);

Hero.defaultProps = {
  content: {
    background: {name: "homepage-h-d"},
    calltoactionurl: "https://www.wiggle.co.uk/discover-different",
    headline: "Discover different",
    calltoactiontext: "Find out more"
  }
}

export default Hero;
