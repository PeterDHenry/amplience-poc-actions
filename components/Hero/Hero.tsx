import React from 'react'

import HeroImage from '../LazyLoadImage/HeroImage'

//TODO need to create border-${color} bg-${color}-hover text-${contrastColor}-hover classes

export interface HeroProps {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  background: { name: string }
  calltoactionurl: string
  calltoactiontext: string
  color: string
  contrastColor: string
  cssClass: string
  headline: string
  strapline: string
}

export const Hero = ({
  background,
  calltoactionurl,
  calltoactiontext,
  color,
  contrastColor,
  cssClass,
  headline,
  strapline,
}: HeroProps) => (
  <a className={`d-block position-relative ${cssClass}`} href={calltoactionurl}>
    <HeroImage id={background.name} />
    <div
      className={`position-absolute bottom-0 start-0 p-5 text-uppercase text-${color}`}
    >
      <h2 className="">{headline}</h2>
      {strapline && <h3 className="">{strapline}</h3>}
      <div
        className={`btn border-${color} bg-${color}-hover text-${contrastColor}-hover`}
      >
        {calltoactiontext}
      </div>
    </div>
  </a>
)
export default Hero

Hero.defaultProps = {
  cssClass: '',
  background: { name: 'homepage-h-d' },
  calltoactionurl: 'https://www.wiggle.co.uk/discover-different',
  headline: 'Default headline',
  strapline: '',
  calltoactiontext: 'Default call to action',
  color: 'white',
  contrastColor: 'black',
}
