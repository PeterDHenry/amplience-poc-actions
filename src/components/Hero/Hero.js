import React from 'react';

import HeroImage from '../HeroImage/HeroImage';

const Hero = ({ content }) => (
  <a className="relative block group" href= {content.calltoactionurl}>
    <HeroImage id={content.background.name} />
    <div className="absolute bottom-40 left-20 uppercase text-white max-w-xs">
      <h3 className="text-7xl font-bold">
        {content.headline}
      </h3>
      <div className="max-w-xs border-solid border-2 border-white p-1 group-hover:bg-white group-hover:text-black">
        {content.calltoactiontext}
      </div>
    </div>
  </a>
);
export default Hero;
