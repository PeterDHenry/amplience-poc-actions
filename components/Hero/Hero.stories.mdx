import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'
import Hero from '../Hero/Hero'

<Meta
  title="CMS Content/Hero"
  component={Hero}
  argTypes={{
    color: {
      options: [
        'black',
        'white',
        'red-500',
        'green-500',
        'yellow-500',
        'blue-500',
      ],
      control: { type: 'select' },
    },
    contrastColor: {
      options: [
        'black',
        'white',
        'red-500',
        'green-500',
        'yellow-500',
        'blue-500',
      ],
      control: { type: 'select' },
    },
  }}
  args={{
    color: 'white',
    contrastColor: 'black',
  }}
/>

export const Template = (args) => <Hero {...args} />

<Canvas>
  <Story name="Default">{Template.bind({})}</Story>
</Canvas>
