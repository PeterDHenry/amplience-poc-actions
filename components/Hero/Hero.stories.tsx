import { Meta, Story } from '@storybook/react'
import { Hero as Component } from './Hero'
import { HeroProps, defaultProps } from './types'

const meta: Meta = {
  title: 'CMS Content/Banners',
  component: Component,
  argTypes: {
    color: {
      options: [
        'black',
        'white',
        'red-500',
        'green-500',
        'yellow-500',
        'blue-500',
      ],
      control: { type: 'select' },
    },
    contrastColor: {
      options: [
        'black',
        'white',
        'red-500',
        'green-500',
        'yellow-500',
        'blue-500',
      ],
      control: { type: 'select' },
    },
  },
}

export default meta

type StoryProps = HeroProps

export const Hero: Story<StoryProps> = (props) => <Component {...props} />
Hero.args = defaultProps
