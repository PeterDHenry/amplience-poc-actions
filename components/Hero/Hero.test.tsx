import { render } from '@testing-library/react'
import Hero, { HeroProps } from './Hero'

describe('Hero component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Hero {...heroProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})

const heroProps: HeroProps = {
  _meta: {
    name: 'string',
    schema: 'string',
    deliveryId: '1543512973',
  },
  cssClass: '',
  background: { name: 'homepage-h-d' },
  calltoactionurl: 'https://www.wiggle.co.uk/discover-different',
  headline: 'Default headline',
  strapline: 'Default Strapline',
  calltoactiontext: 'Default call to action',
  color: 'white',
  contrastColor: 'dark',
}