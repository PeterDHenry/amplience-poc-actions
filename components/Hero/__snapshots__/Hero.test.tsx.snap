// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="Hero__OuterLink-sc-1h6z6u8-0 ejkXYN d-block position-relative undefined"
    href="https://www.wiggle.co.uk/pro-riding-for-summer"
  >
    <div
      style="position: relative; display: block; padding-bottom: 33.85416666666667%;"
    >
      <noscript />
      <img
        alt="Default hero alt text"
        data-src="https://cdn.media.amplience.net/i/wigglecrcnonprd/CCC-11-5-a?w=1920&h=650&sm=C"
        src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
        style="height: 100%; width: 100%; visibility: hidden; position: absolute; top: 0px; left: 0px;"
      />
      <div
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background: white; transition: 0.1s opacity;"
      />
    </div>
    <div
      class="container"
    >
      <div
        class="position-relative"
      >
        <div
          class="position-absolute bottom-0 pb-5 mb-5 text-uppercase"
        >
          <h2
            class=""
          >
            Default  Ride like a pro
          </h2>
          <div
            class="btn"
          >
            Default Shop now
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
